### PR TITLE
[ADD] commodity_trading: module for commodity trading

### PR DESCRIPTION
**Pull request by paso-odoo**
Tailored for commodity traders (e.g., soybeans, wheat) with comprehensive functions, including CRM, quality checks, and broker involvement for product quality and commissions.

task-3460474

### CODE REVIEW

File: commodity_trading/data/mrp_routing_workcenter.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
```
Comment (paso-odoo) [inline commodity_trading/data/mrp_routing_workcenter.xml:2]:
```suggestion
<odoo noupdate="1">
```

File: commodity_trading/data/res_config_settings.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="res_config_settings_multi_step_routes" model="res.config.settings">
```
Comment (paso-odoo) [inline commodity_trading/data/res_config_settings.xml:3]:
```suggestion
    <record id="res_config_settings_commodity_trading" model="res.config.settings">
```

File: commodity_trading/data/res_config_settings.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        <field name="group_product_variant" eval="0"/>
```
Comment (paso-odoo) [inline commodity_trading/data/res_config_settings.xml:4]: bydefault it is disabled?

File: commodity_trading/data/res_config_settings.xml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        <field name="group_stock_packaging" eval="1"/>
+        <field name="group_sale_pricelist" eval="1"/>
+        <field name="group_stock_tracking_lot" eval="1"/>
+        <field name="group_stock_picking_wave" eval="1"/>
+        <field name="use_invoice_terms" eval="1"/>
+        <field name="group_mrp_routings" eval="1"/>
+    </record>
+
+    <function model="res.config.settings" name="execute">
+        <value eval="[ref('res_config_settings_multi_step_routes')]"/>
```
Comment (paso-odoo) [inline commodity_trading/data/res_config_settings.xml:15]:
```suggestion
        <value eval="[ref('res_config_settings_commodity_trading')]"/>
```

File: commodity_trading/data/worksheet_template.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="worksheet_template_1" model="worksheet.template" context="{'worksheet_no_generation': True}">
+        <field name="name">Inward checking</field>
```
Comment (paso-odoo) [inline commodity_trading/data/worksheet_template.xml:4]:
```suggestion
        <field name="name">Inward Checking</field>
```

File: commodity_trading/demo/mrp_production.xml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="mrp_production_1" model="mrp.production">
+        <field name="product_id" ref="product_product_8"/>
+        <field name="product_qty">20.0</field>
+    </record>
+    <record id="mrp_production_2" model="mrp.production">
+         <field name="product_qty">27.0</field>
+         <field name="product_id" ref="product_product_8"/>
```
Comment (paso-odoo) [inline commodity_trading/demo/mrp_production.xml:9]: sequence should be same as above record

File: commodity_trading/demo/purchase_order.xml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        <field name="x_broker" ref="res_partner_9"/>
+    </record>
+    <record id="purchase_order_7" model="purchase.order">
+        <field name="partner_id" ref="res_partner_7"/>
+        <field name="x_broker" ref="res_partner_9"/>
+    </record>
+    <record id="purchase_order_8" model="purchase.order">
+        <field name="partner_id" ref="res_partner_7"/>
+        <field name="x_broker" ref="res_partner_9"/>
+    </record>
```
Comment (paso-odoo) [inline commodity_trading/demo/purchase_order.xml:34]: I would add user_id

File: commodity_trading/demo/purchase_order_line.xml [hunk 8]
```diff
@@ -0,0 +1,49 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="purchase_order_line_14" model="purchase.order.line">
+        <field name="product_id" ref="product_product_1"/>
+        <field name="order_id" ref="purchase_order_1"/>
+        <field name="product_qty">125.0</field>
+        <field name="price_unit">123.0</field>
```
Comment (paso-odoo) [inline commodity_trading/demo/purchase_order_line.xml:7]: price_unit comes should be set in the product. If already set, we need to remove it

File: commodity_trading/demo/sale_order.xml [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+	</record>
+	<record id="sale_order_7" model="sale.order">
+		<field name="partner_id" ref="res_partner_8"/>
+	</record>
+	<record id="sale_order_8" model="sale.order">
+		<field name="partner_id" ref="res_partner_8"/>
+	</record>
+	<record id="sale_order_9" model="sale.order">
+		<field name="partner_id" ref="res_partner_8"/>
+	</record>
```
Comment (paso-odoo) [inline commodity_trading/demo/sale_order.xml:29]: I would add user_id 
Also need to correct the indentation

File: commodity_trading/demo/sale_order_line.xml [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    <record id="sale_order_line_3" model="sale.order.line">
+        <field name="product_id" ref="product_product_4"/>
+        <field name="order_id" ref="sale_order_3"/>
+        <field name="product_uom_qty">27.0</field>
+    </record>
+    <record id="sale_order_line_4" model="sale.order.line">
+        <field name="product_id" ref="product_product_8"/>
+        <field name="order_id" ref="sale_order_4"/>
+        <field name="product_uom_qty">27.0</field>
+        <field name="price_unit">123.0</field>
```
Comment (paso-odoo) [inline commodity_trading/demo/sale_order_line.xml:22]: price_unit should be configured in product itself.

File: commodity_trading/demo/sale_order_post.xml [hunk 11]
```diff
@@ -0,0 +1,131 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
+
```
Comment (paso-odoo) [inline commodity_trading/demo/sale_order_post.xml:3]:
```suggestion
```

File: commodity_trading/demo/sale_order_post.xml [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+        }"/>
+    </function>
+
+    <function name="action_done" model="stock.picking.batch">
+        <value model="stock.picking.batch" eval="obj().env.ref('commodity_trading.sale_order_2').picking_ids.batch_id.id"/>
+    </function>
+    <function name="action_done" model="stock.picking.batch">
+        <value model="stock.picking.batch" eval="obj().env.ref('commodity_trading.sale_order_5').picking_ids.batch_id.id"/>
+    </function>
+
```
Comment (paso-odoo) [inline commodity_trading/demo/sale_order_post.xml:130]:
```suggestion
```

**Review by paso-odoo — COMMENTED** [review]
@dhrs-odoo 
Thanks for working on this PR.

Here are some technical comments.
Cheers!

File: commodity_trading/data/res_config_settings.xml (hunk 3 above)
Comment (dhrs-odoo) [reply commodity_trading/data/res_config_settings.xml:4]: By default it is enabled and as per the db it is not enabled so we have made it disabled

**Comment by dhrs-odoo** [comment]
Hello @paso-odoo 
Thanks for the suggestions 
Given changes has been implemented 
Thank You :innocent:

**Pull request closed without merging**